dsp-shared/src/main/scala/dsp/util/UuidGenerator.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright © 2021 - 2022 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package dsp.util
+
+import zio._
+
+import java.util.UUID
+
+/**
+ * Handles UUID creation
+ */
+trait UuidGenerator {
+  def createRandomUuid: UIO[UUID]
+}
+
+/**
+ * Live instance of the UuidGenerator
+ */
+final case class UuidGeneratorLive() extends UuidGenerator {
+
+  /**
+   * Creates a random UUID
+   *
+   * @return a random UUID
+   */
+  override def createRandomUuid: UIO[UUID] = ZIO.succeed(UUID.randomUUID())
+}
+object UuidGeneratorLive {
+  val layer: ULayer[UuidGeneratorLive] =
+    ZLayer.succeed(UuidGeneratorLive())
+}
+
+/**
+ * Test instance of the UuidGenerator which stores the created UUIDs, so that
+ * they can be queried / returned in tests
+ *
+ * @param listOfUuids the list of created UUIDs
+ */
+final case class UuidGeneratorTest(listOfUuids: Ref[List[UUID]]) extends UuidGenerator {
```

Comment:
I would probably rename it to something like `MockUuidGenerator` and also put it into the `src/test/scala`source set as this would prevent us from accidentially use it in production code.

Reply:
I renamed it to `UuidGeneratorMock` according to our (inofficial?) naming pattern to put "Live", "Test", "Mock" etc. at the end.